### PR TITLE
FH-3746 - Change the cmds  eventalert * to follow the V3 standard and…

### PR DESCRIPTION
**Pull request by camilamacedo86**
JIRA: https://issues.jboss.org/browse/FH-3746

**Following the changes**

- Move to V3 format and directory
- Add unit tests
- Improve output
- Change to use the same REST API of the NGUI

**Steps to Verify**

- Create a new project
- Go to the cloud app of this project
- Go to `Notifications` tab
- Click on Alerts ( this command is about this area )
- Create a new alert via the studio
- List all alerts via fhc 
Command: `fhc eventalert list --app=<app> --env=<environment>`
- Clone a alert
Command: `fhc eventalert clone --app=<app> --id=<id> --name=<name> --env=<environment>`
- Delete a alert
Command: `fhc eventalert delete --app=<app> --id=<id>  --env=<environment>`
- Read a alert
Command: `fhc eventalert read --app=<app> --id=<id> --env=<environment>`
- Update a alert **
Command: `fhc eventalert update --app=<app> --id=<id> --name=<name> --categories=<categories> --severities=<severities> --events=<events> --emails=<emails> --env=<environment>`
- Create a alert **
Command: `cmd : 'fhc eventalert create --app=<app> --name=<name> --categories=<categories> --severities=<severities> --events=<events> --emails=<emails> --env=<environment>`

**IMPORTANT** Use valid values for `--categories=APP_STATE,APP_ENVIRONMENT --severities=INFO,WARN,ERROR,FATAL --events=CRASHED`

If an invalid value is used the following error is found and the data is compromised as follows.

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert create 7gwbu7lepnxijuyhcxrqtodg --name=New22 --categories=APP_STATE,APP_ENVIRONMENT --severities=INFO,WARN,ERROR,FATAL --events=CRAD --emails=emails@test.com,emails2@test.com --env=dev
fhc ERR! String index out of range: -1 
fhc Command executed with error
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert list 7gwbu7lepnxijuyhcxrqtodg --env=dev 
{
  "error": "String index out of range: -1",
  "status": "error"
}
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
``` 

A JIRA was open for the REST API check the informed values before save. The JIRA is: https://issues.jboss.org/browse/RHMAP-16656

**Following the local tests**

- To create:

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert create cnrpzc7ssngewe37parb5m7a --name=Test2 --categories=APP_STATE,APP_ENVIRONMENT --severities=INFO,WARN,ERROR,FATAL --events=CRASHED --emails=emails@test.com,emails2@test.com --env=dev --json
{
  "alertName": "Test2",
  "emails": "emails@test.com,emails2@test.com",
  "enabled": true,
  "env": "dev",
  "eventCategories": "APP_STATE,APP_ENVIRONMENT",
  "eventNames": "CRASHED",
  "eventSeverities": "INFO,WARN,ERROR,FATAL",
  "guid": "f3bcw4jwsioqid37gewqjeoc",
  "status": "ok",
  "sysCreated": "2017-07-06 18:23:13",
  "sysModified": "2017-07-06 18:23:13",
  "uid": "cnrpzc7ssngewe37parb5m7a"
}
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```

screen shot 2017-07-06 at 3 23 18 pm

screen shot 2017-07-06 at 3 24 28 pm

- To List:

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert list cnrpzc7ssngewe37parb5m7a --env=dev --json
{
  "list": [
    {
      "alertName": "Test2",
      "emails": "emails@test.com,emails2@test.com",
      "enabled": true,
      "env": "dev",
      "eventCategories": "APP_STATE,APP_ENVIRONMENT",
      "eventNames": "CRASHED",
      "eventSeverities": "INFO,WARN,ERROR,FATAL",
      "guid": "f3bcw4jwsioqid37gewqjeoc",
      "sysCreated": "2017-07-06 18:23:13",
      "sysModified": "2017-07-06 18:23:13",
      "uid": "cnrpzc7ssngewe37parb5m7a"
    },
    {
      "alertName": "Test",
      "emails": "emails@test.com,emails2@test.com",
      "enabled": true,
      "env": "dev",
      "eventCategories": "APP_STATE,APP_ENVIRONMENT",
      "eventNames": "CRASHED",
      "eventSeverities": "INFO,WARN,ERROR,FATAL",
      "guid": "ndewinnsbpk25cdwsbj62cok",
      "sysCreated": "2017-07-06 18:22:54",
      "sysModified": "2017-07-06 18:22:54",
      "uid": "cnrpzc7ssngewe37parb5m7a"
    }
  ],
  "status": "ok"
}
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```
screen shot 2017-07-06 at 3 25 32 pm

- To delete:

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert delete cnrpzc7ssngewe37parb5m7a ndewinnsbpk25cdwsbj62cok --env=dev --json
{
  "alertName": "Test",
  "emails": "emails@test.com,emails2@test.com",
  "enabled": true,
  "env": "dev",
  "eventCategories": "APP_STATE,APP_ENVIRONMENT",
  "eventNames": "CRASHED",
  "eventSeverities": "INFO,WARN,ERROR,FATAL",
  "guid": "ndewinnsbpk25cdwsbj62cok",
  "status": "ok",
  "sysCreated": "2017-07-06 18:22:54",
  "sysModified": "2017-07-06 18:22:54",
  "uid": "cnrpzc7ssngewe37parb5m7a"
}
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert delete cnrpzc7ssngewe37parb5m7a f3bcw4jwsioqid37gewqjeoc --env=dev
Event alerts with the ID 'f3bcw4jwsioqid37gewqjeoc' for the app 'cnrpzc7ssngewe37parb5m7a' into the env 'dev' deleted successfully.
```

- To update: 

screen shot 2017-07-06 at 3 44 10 pm

- To clone:

screen shot 2017-07-06 at 3 45 59 pm

### CODE REVIEW

File: lib/cmd/fh3/eventalert/clone.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  'describe' : {
+    'app'  : i18n._("Unique 24 character GUID of your cloud application."),
+    'id'   : i18n._("Unique 24 character GUID of the event alert"),
+    'name'   : i18n._("Name of the event alert"),
+    'env'  : i18n._("Unique 24 character GUID of the environment where this application is deployed."),
+    'json' : i18n._("Output in json format")
+  },
+  'customCmd': function(argv, cb) {
+    fhc.eventalert.read({app: argv.app, env: argv.env, id:argv.id, json:true}, function(err, alert) {
```
Comment (wei-lee) [inline lib/cmd/fh3/eventalert/clone.js]: I think only `id` should be required? At least I don't think `env` is required here.

File: lib/cmd/fh3/eventalert/create.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    'name'   : i18n._("Name of the event alert"),
+    'categories' : i18n._("Categories of this alert. (E.g APP_STATE,APP_ENVIRONMENT"),
+    'severities' : i18n._("Severities of this alert. (E.g INFO,WARN,ERROR,FATAL"),
+    'events' : i18n._("Severities of this alert. (E.g START_REQUESTED,DEPLOY_REQUESTED"),
+    'emails' : i18n._("Emails which will received this alert. (E.g test@test.com, test2@test.com"),
+    'env'   : i18n._("Unique 24 character GUID of the environment where this application is deployed."),
+    'json' : i18n._("Output in json format")
+  },
+  'customCmd': function(argv, cb) {
+    var url =  "/api/v2/mbaas/support/" + argv.env + "/apps/" + argv.app + "/alerts";
```
Comment (wei-lee) [inline lib/cmd/fh3/eventalert/create.js]: where is this url come from? Don't remember I have seen it before.

File: lib/cmd/fh3/eventalert/clone.js (hunk 1 above)
Comment (wei-lee) [inline lib/cmd/fh3/eventalert/clone.js]: where is `fhc.eventalert` defined?

File: lib/cmd/fh3/eventalert/create.js (hunk 2 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/eventalert/create.js]: It is the REST service used by NGUI FH3 and I believe that URL which was before is from the legacy.

File: lib/cmd/fh3/eventalert/clone.js (hunk 1 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/eventalert/clone.js]: 👍 I will remove the env and put the name as Copy+OldName.
About the second comment, fhc.eventalert is == lib/cmd/fh3/eventalert/

**Comment by camilamacedo86** [comment]
@wei-lee Really thank you for your review. I answered the questions made. Could you check if we can move forward with this one?

**Pull request merged**